src/ZohoCopyDatabaseCommand.php

```diff
@@ -27,36 +28,72 @@ class ZohoCopyDatabaseCommand extends Command
-    private $lock;
+    private $lockSync;
 
     /**
-     * @param ZohoDatabaseCopier                $zohoDatabaseCopier
-     * @param \Wabel\Zoho\CRM\AbstractZohoDao[] $zohoDaos           The list of Zoho DAOs to copy
-     * @param Lock                              $lock               A lock that can be used to avoid running the same command twice at the same time
+     * 
+     * @param \Wabel\Zoho\CRM\Copy\ZohoDatabaseCopier $zohoDatabaseCopier
+     * @param \Wabel\Zoho\CRM\Copy\ZohoDatabaseSyncZoho $zohoDatabaseSync
+     * @param array $zohoDaos The list of Zoho DAOs to copy
+     * @param Lock $lockCopy A lock that can be used to avoid running the same command (copy) twice at the same time
+     * @param Lock $lockSync A lock that can be used to avoid running the same command (sync) twice at the same time
      */
-    public function __construct(ZohoDatabaseCopier $zohoDatabaseCopier, array $zohoDaos, Lock $lock = null)
+    public function __construct(ZohoDatabaseCopier $zohoDatabaseCopier, ZohoDatabaseSyncZoho $zohoDatabaseSync, array $zohoDaos, Lock $lockCopy = null, Lock $lockSync = null)
     {
         parent::__construct();
         $this->zohoDatabaseCopier = $zohoDatabaseCopier;
+        $this->zohoDatabaseSync = $zohoDatabaseSync;
         $this->zohoDaos = $zohoDaos;
-        $this->lock = $lock;
+        $this->lockCopy = $lockCopy;
+        $this->lockSync = $lockSync;
     }
 
     protected function configure()
     {
         $this
             ->setName('zoho:copy-db')
-            ->setDescription('Copies the Zoho database in local DB tables')
-            ->addOption("reset", "r", InputOption::VALUE_NONE, 'Get a fresh copy of Zoho (rather than doing incremental copy)');
+            ->setDescription('Copies the Zoho database in local DB tables and synchronize Zoho CRM from the Zoho database.')
+            ->addArgument("action",  InputArgument::REQUIRED, "Specify 'copy' or 'sync'")
+            ->addOption("reset", "r", InputOption::VALUE_NONE, 'Get a fresh copy of Zoho (rather than doing incremental copy)')
+            ->addOption("trigger", "t", InputOption::VALUE_NONE, 'Create or update the triggers');
     }
 
     protected function execute(InputInterface $input, OutputInterface $output)
     {
+        $action = $input->getArgument('action');
+        switch ($action) {
+            case 'copy':
+                $this->copyDb($input, $output);
+                break;
+            case 'sync':
+                $this->syncDb($output);
+                break;
+            default:
+                throw new InvalidArgumentException('Named argument not found.');
+        }
+    }
+
+    /**
+     * Run the copy Db command.
+     * @param InputInterface $input
+     * @param OutputInterface $output
+     */
+    private function copyDb(InputInterface $input, OutputInterface $output){
         try {
-            if ($this->lock) {
-                $this->lock->acquireLock();
+            if ($this->lockCopy) {
```

Comment:
We have 2 different locks for the 2 actions. It means that technically, it is possible to trigger a copy and a sync at the same time. Couldn't it be dangerous? Maybe it is safer to put a same lock for both actions?